### ISSUE TITLE
New note type after fullsync is created wrongly

### ISSUE THREAD

**Issue by rgamici** [ankidroid/Anki-Android#5450]
###### Reproduction Steps

1. Do a full sync of the collection
2. Create a new note type (Add: Basic (and reversed card) or Basic (with optional reversed card))

###### Expected Result

A note type with two cards should be created.

###### Actual Result

The note type list in not refreshed to show the new one.
When the note type list is open again, the new one has only the front card.

It also ignores the user inputted name.
The random suffix (to avoid note types with the same name) does not appear.
Trying a complete different name also fails.
In both cases the new note type name is "Basic (and reversed card)".

I also tried to a normal sync after creating the defective note type, but the problem persisted.

After the defective note type is removed (require a full sync later), if a new one is created, this time it will be correct.
The note type list is also correctly refreshed after the creation.
The inputted name is respected.

###### Debug info
Ankidroid version: 2.9Beta12
Android version: 5.1.1
Acra uuid = a78b6ca1-0fec-41c0-8b0a-702d0eca9ba0

###### Research
*Enter an [ x ] character to confirm the points below:*

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

**Comment by timrae** [comment]
Ouch. I don't really have time to look at this right now, but seems important to figure out what's going on.

**Comment by rgamici** [comment]
I think that problem appears from time to time on the mail list.
Can we have at least a new entry on the FAQ saying that the bug exist and
how deal with it?
It could help those who checked the FAQ or at least make it easier to reply
to those question by linking to the FAQ.

**Comment by timrae** [comment]
Sure, FAQ can be edited by anyone, please go ahead!

**Comment by timrae** [comment]
Can you upload your collection just after step one and then just after step two?

**Comment by rgamici** [comment]
I don't want to upload my personal collection, but I was able to reproduce it with an empty collection, which was created by changing the ankidroid folder.

The steps I did were:
1. Delete old note types
2. Add a new card (I could not full sync with no cards)
3. Do a full-sync upload
4. Export the collection (named as step 1)
5. Create a new note type for reversed cards (it failed to show it as in the original post)
6. Export the collection (named as step 2)
7. Confirmed that the note type created was wrong, there were just one card, the front one.

I also noticed that when the collection was created by assigning a new folder, it comes with several note types (reversed, controlled reversed), which were OK.
After I deleted them and tried a normal sync (or even a full sync before adding a card (it automatically downloaded from Ankiweb)), the note types were recreated automatically.
After doing full sync upload, they were not created again anymore.

I noticed that there is a note type called "Basic (type in the answers)" which does not appear on the list of new card templates.
It was also not available when the new collection was created, and it appears only after the first sync.
When I start a new collection (without the type in template) and add a card, a full sync is required.

[collection.zip](https://github.com/ankidroid/Anki-Android/files/3726473/collection.zip)

**Comment by snoyes** [comment]
Basic and Cloze work because they add the card templates to the new model before adding the model to the collection.

Basic ([optional|and] reversed card) reuse the function to create a Basic card (which adds the first template and adds the model to the collection, which assigns it an id), but then they add a second template. That addTemplate call throws a ConfirmModSchemaException because the id != 0 anymore. However, that exception is not handled:
https://github.com/ankidroid/Anki-Android/blob/bcc13aaf374725af69159bc4c7f68ea0d2fa3ea9/AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.java#L422

**Comment by timrae** [comment]
This should be fixed in v2.10alpha6. Can you guys please test?

**Comment by snoyes** [comment]
Looks resolved to me - honors the new name, does not require an overwrite  to sync.
